src/storage/transaction/ChainUpdateProcessor.cpp

```diff
@@ -0,0 +1,117 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "storage/transaction/ChainUpdateProcessor.h"
+#include "storage/transaction/TransactionUtils.h"
+
+namespace nebula {
+namespace storage {
+
+folly::SemiFuture<cpp2::ErrorCode> ChainUpdateEdgeProcessor::prepareLocal() {
+    LOG_IF(INFO, FLAGS_trace_toss) << "prepareLocal(), txnId_=" << txnId_;
+    sLockKey_ = TransactionUtils::lockKey(vIdLen_, partId_, inEdgeKey_);
+    lk_ = std::make_unique<LockGuard>(env_->txnMan_->getMemoryLock(spaceId_), sLockKey_, txnId_);
```

Comment:
The last byte of lock in TransactionUtils::lockKey is always 0, so for one in-edge, there can only be one lock?

For example, there is already a lock on the current in-edge, but it will never be locked later?

Reply:
yes, there will be only one lock.